keps/sig-storage/20191117-immutable-secrets-configmaps.md

```diff
@@ -128,11 +128,16 @@ field:
 ```
 
 If set, the machinery in apiserver will reject any updates of the object
-trying to change anything different than ObjectMetadata. Note that we
-need to allow mutating ObjectMetadata to not break object lifecycle
-(e.g. to avoid deadlock if Finalizers are set).
+trying to change anything different than ObjectMetadata.
 
-Based on the value of that field Kubelet will or will not:
+Note that will NOT reject all updates of the object, as we need to allow
+e.g. for mutating ObjectMetadata (to not break object lifecycle, e.g. by
+introducing a deadlock if Finalizers are set) or to allow rotating
+certificates used for encryption at rest. We will only reject requests
+that are explicitly changing keys and/or values stored in Secrets and/or
+ConfigMaps.
+
+Based on the value of `Immutable` field Kubelet will or will not:
 - start a watch (or periodic polling) of a given Secret/ConfigMap
 - perform updates of files mounted to a Pod based on updates of
```

Comment:
I might be wrong, but is volume remount is triggered by pod update, not volume update? Does volume (secret) update will trigger pod update?
https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/volumemanager/cache/actual_state_of_world.go#L69

Reply:
Yes - we do update them periodically:
https://github.com/kubernetes/kubernetes/blob/cfdfd043a086eba12a673faf7a7dee89d28c82e3/pkg/volume/secret/secret.go#L85

Reply:
Secret volume (and configmap volume) is mounted read-only in the container, so users cannot update the volume contents.

The fact that the markremount function is in asw is wrong and it should be refactored eventually to dsw: https://github.com/kubernetes/kubernetes/issues/77758